search-commons/src/main/java/no/unit/nva/search/resource/ContributorNodeFilter.java

```diff
@@ -0,0 +1,57 @@
+package no.unit.nva.search.resource;
+
+import static no.unit.nva.constants.Words.AFFILIATIONS;
+import static no.unit.nva.constants.Words.COUNTRY_CODE;
+import static no.unit.nva.constants.Words.DOT;
+import static no.unit.nva.constants.Words.ENTITY_DESCRIPTION_CONTRIBUTORS_PATH;
+import static no.unit.nva.constants.Words.IDENTITY;
+import static no.unit.nva.constants.Words.NO;
+import static no.unit.nva.constants.Words.VERIFICATION_STATUS;
+import static no.unit.nva.constants.Words.VERIFIED;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import no.unit.nva.search.common.records.JsonNodeMutator;
+
+public final class ContributorNodeFilter implements JsonNodeMutator {
+
+    static final int FIRST_FIVE_CONTRIBUTORS = 5;
+    static final JsonPointer CONTRIBUTORS_PATH_POINTER =
+            JsonPointer.compile(ENTITY_DESCRIPTION_CONTRIBUTORS_PATH);
+
+    private ContributorNodeFilter() {}
+
+    public static ContributorNodeFilter verifiedOrNorwegian() {
+        return new ContributorNodeFilter();
+    }
+
+    @Override
+    public JsonNode transform(JsonNode source) {
+        var target = source.at(CONTRIBUTORS_PATH_POINTER);
+        var elements = target.elements();
+        while (elements.hasNext()) {
+            var element = elements.next();
+            if (deleteAfter(element, FIRST_FIVE_CONTRIBUTORS)
+                    && isNotVerified(element)
+                    && hasNoNorwegianBasedContributor(element)) {
```

Comment:
Try shortering the transform() method by putting the 3 delete-checks in a shouldDeleteContributor() function